[ATen-CPU] Add `math.h` for Gelu

SS-JIA opened the pull request:
Summary:
## Context

This PR is mostly to enable ExecuTorch build for Windows: https://github.com/pytorch/executorch/pull/9198

In ExecuTorch, the optimized GeLU kernel calls the ATen implementation. However, on Windows `math.h` needs to be included with `#define _USE_MATH_DEFINES` in order for math constants to be defined.

Test Plan:
Rely on CI to make sure existing tests do not break. Tested separately with ExecuTorch to make sure Windows build is successful.


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

swolchok commented on aten/src/ATen/native/cpu/Gelu.h:
```diff
@@ -1,5 +1,12 @@
+// access constants such as M_SQRT2 and M_2_SQRTPI.
+#ifdef _WIN32
+#define _USE_MATH_DEFINES
+#include <math.h>
```
why not include this header unconditionally?

also can we use <cmath> instead?

SS-JIA replied:
I mainly wanted to limit the scope of the changes to Windows only builds.

> also can we use instead?

Sorry, I don't understand this comment 😅

SS-JIA commented:
`<cmath>` seems to trigger `lintrunner` failures: https://github.com/pytorch/pytorch/actions/runs/13846450782/job/38745821504?pr=149164

Error message snippet:

```
>>> Lint for ../../usr/include/c++/11/cmath:

  Error (CLANGTIDY) [clang-diagnostic-error]
    constexpr function 'fpclassify' without __host__ or __device__ attributes
    cannot overload __device__ function with the same signature; add a
    __host__ attribute, or build with -fno-cuda-host-device-constexpr

         534  |
         535  |#ifndef __CORRECT_ISO_CPP11_MATH_H_PROTO_FP
         536  |  constexpr int
    >>>  537  |  fpclassify(float __x)
         538  |  { return __builtin_fpclassify(FP_NAN, FP_INFINITE, FP_NORMAL,
         539  |				FP_SUBNORMAL, FP_ZERO, __x); }
         540  |

  Error (CLANGTIDY) [clang-diagnostic-error]
    constexpr function 'fpclassify' without __host__ or __device__ attributes
    cannot overload __device__ function with the same signature; add a
    __host__ attribute, or build with -fno-cuda-host-device-constexpr

         539  |				FP_SUBNORMAL, FP_ZERO, __x); }
         540  |
         541  |  constexpr int
    >>>  542  |  fpclassify(double __x)
         543  |  { return __builtin_fpclassify(FP_NAN, FP_INFINITE, FP_NORMAL,
         544  |				FP_SUBNORMAL, FP_ZERO, __x); }
         545  |

...
```

Will just hide the include in the `ifdef` block for now...

SS-JIA committed:
[ATen-CPU] Add `math.h` for Gelu

Summary:
## Context

This PR is mostly to enable ExecuTorch build for Windows: https://github.com/pytorch/executorch/pull/9198

In ExecuTorch, the optimized GeLU kernel calls the ATen implementation. However, on Windows `math.h` needs to be included with `#define _USE_MATH_DEFINES` in order for math constants to be defined.

Test Plan:
Rely on CI to make sure existing tests do not break. Tested separately with ExecuTorch to make sure Windows build is successful.

SS-JIA commented:
@pytorchbot merge

SS-JIA commented:
@pytorchbot merge

SS-JIA commented:
@pytorchbot merge -f "The bot is complaining about BC Lint, but the job keeps getting cancelled immediately. I don't think the signal is accurate. All other jobs are green."
